modules/core/src/scenegraph/group.js

```diff
@@ -38,31 +38,18 @@ export default class Group extends Node {
     return this;
   }
 
-  // If visitor returns a truthy value, traversal will be aborted and that value
-  // will be returned from `traverse`. Otherwise `traverse` will return null.
-  traverse(visitor, {modelMatrix: modelMatrixIn = new Matrix4()} = {}) {
+  traverse(visitor, {worldMatrix = new Matrix4()} = {}) {
+    const modelMatrix = new Matrix4(worldMatrix).multiplyRight(this.matrix);
+
     for (const child of this.children) {
-      const {matrix} = child;
-      const modelMatrix = new Matrix4(modelMatrixIn).multiplyRight(matrix);
-      let result;
```

Comment:
Any chance we can restore the return logic now that we fixed the traversal?

We will need a couple of test cases for the scenegraph stuff soon.

Reply:
❌